daemon/algod/api/server/v2/handlers.go

```diff
@@ -785,6 +798,63 @@ func (v2 *Handlers) RawTransaction(ctx echo.Context) error {
 	return ctx.JSON(http.StatusOK, generated.PostTransactionsResponse{TxId: txid.String()})
 }
 
+// SimulateTransaction simulates broadcasting a raw transaction to the network, returning relevant simulation results.
+// (POST /v2/transactions/simulate)
+func (v2 *Handlers) SimulateTransaction(ctx echo.Context) error {
+	if !v2.Node.Config().EnableTransactionSimulator {
+		return ctx.String(http.StatusNotFound, "/transactions/simulate was not enabled in the configuration file by setting EnableTransactionSimulator to true")
+	}
+
+	stat, err := v2.Node.Status()
+	if err != nil {
+		return internalError(ctx, err, errFailedRetrievingNodeStatus, v2.Log)
+	}
+	if stat.Catchpoint != "" {
+		// node is currently catching up to the requested catchpoint.
+		return serviceUnavailable(ctx, fmt.Errorf("SimulateTransaction failed as the node was catchpoint catchuping"), errOperationNotAvailableDuringCatchup, v2.Log)
+	}
+	proto := config.Consensus[stat.LastVersion]
+
+	txgroup, err := decodeTxGroup(ctx, proto.MaxTxGroupSize)
+	if err != nil {
+		return badRequest(ctx, err, err.Error(), v2.Log)
+	}
+
+	actualLedger := v2.Node.LedgerForAPI()
+
+	// Fetch previous block header just once to prevent racing with network
+	hdr, err := actualLedger.BlockHdr(actualLedger.Latest())
+	if err != nil {
+		return internalError(ctx, err, "current block error", v2.Log)
+	}
+
+	// Simulate transaction
+	simulator := MakeSimulatorFromAPILedger(actualLedger, hdr)
+	result, err := simulator.SimulateSignedTxGroup(txgroup)
+	if err != nil {
+		var invalidTxErr *InvalidTxGroupError
+		var scopedErr *ScopedSimulatorError
+		switch {
+		case errors.As(err, &invalidTxErr):
+			return badRequest(ctx, invalidTxErr, invalidTxErr.Error(), v2.Log)
+		case errors.As(err, &scopedErr):
+			return internalError(ctx, scopedErr, scopedErr.External, v2.Log)
+		default:
+			return internalError(ctx, err, err.Error(), v2.Log)
+		}
+	}
+
+	res := generated.SimulationResponse(result)
+
+	// Return msgpack response
+	msgpack, err := encode(protocol.CodecHandle, &res)
```

Comment:
Seems like `encode()` could go away (I know this isn't your doing). It looks like `codec.EncodeReflect` does what you want, jumping through some extra hoops to try to go faster.
If we *really* care about speed, I suppose we skip the reflection and encode directly with msgp routines.

If we like the convenience of `encode()` let's go all the way, and have it do the wrapping up into an `internalError` which happens after every such call.  We'd also skip the Printf inside - not much point